### PR TITLE
Use Kotlin coroutines to fix ANR when refreshing 'My Courses'

### PR DESCRIPTION
**Pull request by abhijeetviswa**
The first commit migrates `MyCoursesFragment` over to Kotlin. It also bumps Kotlin to 1.4.10 to use the latest and greatest Kotlin features.

A step towards closing #225.

### CODE REVIEW

File: app/src/main/java/crux/bphc/cms/fragments/MyCoursesFragment.kt [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+                            }
+                        }
+                    }
+
+                    val newPartsInSections = courseDataHandler
+                            .isolateNewCourseData(it.courseId, sections)
+                    courseDataHandler.replaceCourseData(it.courseId, sections)
+
+                    realm.close() // let's not forget to do this
+                    if (newPartsInSections.isNotEmpty()) {
```
Comment (aryanarora180) [inline app/src/main/java/crux/bphc/cms/fragments/MyCoursesFragment.kt:241]: You can return a 1 for true and 0 for false here and then later on while "folding" it, directly use the values. It's a minor thing though.

**Review by aryanarora180 — APPROVED** [review]
Good to go!

File: app/src/main/java/crux/bphc/cms/fragments/MyCoursesFragment.kt (hunk 1 above)
Comment (abhijeetviswa) [reply app/src/main/java/crux/bphc/cms/fragments/MyCoursesFragment.kt:241]: That is what I did initially. But that made less sense than using `true` or `false` to indicate that a course has updates.

**Comment by abhijeetviswa** [comment]
> Good to go!

Thanks. Squashed fixups and force pushed. All checks passing, will merge it in now.

File: app/src/main/java/crux/bphc/cms/fragments/MyCoursesFragment.kt (hunk 1 above)
Comment (aryanarora180) [reply app/src/main/java/crux/bphc/cms/fragments/MyCoursesFragment.kt:241]: True, it isn't good readability wise

**Pull request merged**